CMakePresets.json

```diff
@@ -9,7 +9,8 @@
                 "CMAKE_CXX_STANDARD"                : "17",
                 "KokkosTools_ENABLE_EXAMPLES"       : "ON",
                 "KokkosTools_ENABLE_SINGLE"         : "ON",
-                "KokkosTools_ENABLE_MPI"            : "ON"
+                "KokkosTools_ENABLE_MPI"            : "ON",
+                "KokkosTools_ENABLE_PAPI"           : "ON"
```

Comment:
I understand that is the easiest way to get this tested in all builds but I am not sure whether enabling PAPI should be configured as the "default" build.  That is the name of the preset that you changed.

Reply:
(I did not request changes)

Reply:
> At first, I thought it would only make sense to test the `PAPI` connector for CPU-related execution spaces.
> 
> 
> 
> But according to https://github.com/icl-utk-edu/papi/wiki (thought I haven't read the details), `PAPI` claims to also work for GPUs (*e.g.* https://github.com/icl-utk-edu/papi/wiki/PAPI-Component-ROCM).
> 
> 
> 
> I'm not against testing `PAPI` only for the presets like `OpenMP`. I'll do what you advise :wink:

Thanks very much for these changes and thanks for your particular question on PAPI GPU library support.

I absolutely think PAPI should be tested for more than presets like OpenMP. 


Given my own prior experience with PAPI, and recent interactions with Kokkos users and developers, Kokkos should allow for PAPI profiling for GPUs and not just CPUs. With support for PAPI in CUDA for more than a few years and recent maturity of PAPI for HIP over the past year, I think it is fundamentally important that Kokkos Tools do tests on PAPI GPU libraries.

Broadly speaking: Kokkos users should not have to rely on vendor's derived metrics through their visualization tools. They should be able to have direct access to hardware counters. I think this true for any scientific application program using CUDA, HIP, SyCL. 

I don't know if there is a SYCL PAPI.



With all the above said, I think that In principle, putting this particular change on PAPI GPU libraries in a separate Kokkos Tools PR may make more sense so that your non PAPI GPU related can be merged quickly.

 @dalg24 ?